### DISCUSSION TITLE
Drag the route

### DISCUSSION THREAD

**Discussion by Leeb64** [devemux86/cruiser#476] [Q&A] [answered]
With a route that runs more or less in a “zigzag” shape, you often have to first set an intermediate point on the route -> then move it to achieve the desired change.
If you simply place the intermediate point at a desired location, you often don't get the desired result.
An elegant solution would be to "move" the route.
You “click” on a specific spot and drag it to the desired location.

**Comment by devemux86** [comment]
@Leeb64 thanks for the suggestion.

Do you mean like the [BRouter website](https://brouter.de/brouter-web/)?

Although mobile apps with touch gestures are not like desktop websites with mouse control.

**Reply by Leeb64** [reply]
In priciple,Yes.

**Comment by 0709wiwiwi** [comment]
I fully understand the suggestion. I know that this was also once discussed with the Kurviger **PHONE APP.** However, it was decided then to prefer the current cross hair method because it forces a very precise placement of the planner points (shaping or via). It served primarily to prevent rapid sloppiness in planning points placements into **THE PHONE APP** and thus so prevent annoying navigation instructions errors misery.

Still this. 
In Locus, choose or quick dragging or precise positioning.  I prefer precise.  Correcting wrongly positioned points later is extra work.

**Comment by MotoUKRider** [comment]
This sort of feature would work if Cruiser had "road awareness" when placing ShapingPoints - perhaps driven by the active Profile - it could land/snap on a road and name the point accordingly (to use a loose term: "like the other apps do").

I do find the cross-hair method a little too long-winded in most situations but there is a margin for error so you don't have to place them exactly on the road when making routes "in the app".  How precise do you need to be?  If you zoom in far enough, a finger gesture *should* be plenty.

/2c

**Comment by devemux86** [comment]
@Leeb64 probably means drag the line to create an intermediate point, like in [BRouter website](https://brouter.de/brouter-web/).

The previous discussion was about (long press) drag waypoints (losing their menu) vs the current precise crosshair method.

**Reply by 0709wiwiwi** [reply]
I suppose so, as is common with Google and some other (web) planners.  Probably it could work relatively elegantly in, say, the desktop version. On a fairly small phone screen working with high precision by quickly dragging the line while a finger that  covers the map view is less obvious.

@MotoUKRider 
> This sort of feature would work if Cruiser had "road awareness"

Indeed, for road motorbiking, the road selection is already  a bit limited. However, cyclists or pedestrians want to select (bicycle) (side) paths. That additional multiple choice means that the postioning has to be very precise anyway. You must avoid that your route switches between the main road (bikes allowed) or the nearby beautiful forest path and vice versa for example.

**Reply by Leeb64** [reply]
Basically it's about placing and moving an intermediate point at a very specific point on the route.
When I set an intermediate point to move the route there, it happens that the point is approached from another, closer point on the route.
I first have to insert an intermediate point at the location of the existing route that I want to move and then move it.
This is a bit complicated.
It would be a little easier to “drag” the route.
You can also drag with the crosshairs...

**Reply by devemux86** [reply]
> I first have to insert an intermediate point at the location of the existing route that I want to move and then move it.

Can long press anywhere on the map and set via / shaping point on desired location directly in 1 step.
(instead of add point on the route and them move it)

**Reply by Leeb64** [reply]
![Screenshot](https://github.com/devemux86/cruiser/assets/120722053/a853acf9-5d11-495f-856e-e22e70a9cf67)
Because of this!
I first entered WP2 (long the Ardeche), then wanted to enter a detour (WP1).
This is the result.

**Reply by Leeb64** [reply]
The shortest path is calculated.
But I do not want that!

**Reply by Leeb64** [reply]
This was the Plan:
![grafik](https://github.com/devemux86/cruiser/assets/120722053/fbf2cd44-ffe0-4b0f-8559-b7b7046e5bf1)

**Reply by devemux86** [reply]
> I first entered WP2 (long the Ardeche), then wanted to enter a detour (WP1).

The point 1 is automatically ordered as 1st while you wanted it as 2nd?
In such case indeed it's better to first place it on the route and then move it.

**Reply by 0709wiwiwi** [reply]
In any route adjustment, you should check the desired number sequences very carefully.  Remember Undo is your best friend!
Especially If you want to add an extra intermediate straight section to an already existing beeline, you need to pay even more attention to the correct points sequence placements.

See the video. [Create an extra mid beeline track section](https://youtube.com/shorts/vvNuu_3vsL0?feature=share)
  
Adapting an existing beeline which between 1 and 2 is too different from the overlay. There is thus an additional beeline mid section needed by placing a new additional shaping point.  However, if the placement position is closer to 2 than to 1 this becomes undesirable than shaping point 3.  Because of the false number order, this edit will then for sure go all wrong.  
What to do ? Use the undo and restart.  
Place that extra shaping point closer to 1 so it becomes shaping point 2. Always check this carefully. Only then can you move the new created shaping point 2 to the desired correct position.

**Reply by MotoUKRider** [reply]
As @0709wiwiwi says, the example from @Leeb64 is a very good display of Cruiser being *too good* at what it does best: Placing points at the "best" possible place during the route.

The idea of route dragging aside, this workflow does create a question about *how* Cruiser could/should handle manually-placed ShapingPoints: a) Either work out the "best" possible linkage between each point, or b) Strictly obey the *order* in which new points are added and make a corn-maze of the result :laughing: 

I suspect this might involve a "Profile" based solution that determines how to apply the route build, mainly because if I were to plot a 10-mile journey with a ShapingPoint at the 7-mile mark, if I were to add another near the 3-mile mark - the default logic suggests it's where I want to be *earlier* in the route than later.  This sounds like a 2nd glass of wine problem :wine_glass:

**Reply by 0709wiwiwi** [reply]
See the map, the **map scale** and the distances as shown by Leeb. It requires you on a tiny smartphone screen to move that line very precisely. Anyway it seems we fall into repetition ;-)  So find the Kurviger app discussion here: 
https://forum.kurviger.de/t/crosshair-placement-or-drag-waypoints/3152/6?u=0709

**Reply by Leeb64** [reply]
My idea of ​​the process:
I press on the route section that should be moved, the menu opens, I select "Move route section", the crosshairs appear, I move the crosshairs to the appropriate place, confirm!
No more complicated than setting a waypoint, but easier than setting a waypoint and then moving it.
Above all, more intuitive.

**Reply by devemux86** [reply]
> I move the crosshairs to the appropriate place, confirm!

And a waypoint is set there, right?

**Reply by Leeb64** [reply]
YES!
In principle like the BRouter Web Client.

**Comment by devemux86** [comment]
A simple workflow is the following 2 steps:
- Long press the route and select set via or shaping point

- Crosshair appears, move and press OK to set new point

**Reply by Leeb64** [reply]
And what does that do for me?
I'll have to move this point later.

**Reply by devemux86** [reply]
The via / shaping point will not be set immediately, like it happens now.
The crosshair will appear first, move and press OK to add the new point.

So there will be only 1 route calculation without extra movement.

**Reply by devemux86** [reply]
> My idea of ​​the process:
I press on the route section that should be moved, the menu opens, I select "Move route section", the crosshairs appear, I move the crosshairs to the appropriate place, confirm!
No more complicated than setting a waypoint, but easier than setting a waypoint and then moving it.
Above all, more intuitive.

As you described it above, but using the existing set via / shaping point menu.
(without adding extra menu)

**Reply by Leeb64** [reply]
Sometimes I need a little longer...
Of course, you could do it that way too.

**Reply by devemux86** [reply]
> Long press the route and select set via or shaping point
> Crosshair appears, move and press OK to set new point

The new way to set waypoints on route is available for testing in:
- https://github.com/devemux86/cruiser/discussions/482

**Comment by devemux86** [comment, marked as answer]
>Long press the route and select set via or shaping point
Crosshair appears, move and press OK to set new point

The new way to set waypoints on route is available in [Cruiser 4.0.0](https://play.google.com/store/apps/details?id=gr.talent.cruiser).
